### PR TITLE
enable SELinux on android-ia

### PR DESCRIPTION
**Pull request by ceiba1985**
1. Set SELinux enforcing mode in mixins.spec file.
2. Execute mixin.update to update the configure files.
3. Modify fstab to mount config.img on oem_config dir instead
   of config dir.
4. Add the necessary SELinux rules to make the system boot
   up successfully after enabling the SELinux on android-ia.

JIRA: None.
Tests: The system can boot to UI successfully and there are no obvious regressions observed.

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>

### CODE REVIEW

**Comment by tpalli** [comment]
Before committing this please test also that adb remount, adb push works, we need to be able to write anywhere in the disk for development purposes. It would be good to also test dEQP, 3DMark and some other apps that needs to write to the disk and see they still work properly.

**Comment by ceiba1985** [comment]
adb remount, adb push and adb pull all work fine after applying this patch, and there are no obvious regressions observed.

**Comment by ceiba1985** [comment]
I have tried my patch with the userdebug build and there are no obvious regressions found.

Note: To make userdebug build boot up successfully, you have to revert the commit of "Enable default full disk encryption(FDE)".

File: sepolicy/mediaserver.te [hunk 1]
```diff
@@ -8,3 +8,5 @@ allow mediaserver sensorservice_service:service_manager find;
 allow mediaserver host1x_device:chr_file rw_file_perms;
 # needed for widevine classic
 allow mediaserver tee_device:chr_file { ioctl open read write };
+allow mediaserver surfaceflinger_tmpfs:file { read write };
```
Comment (williamcroberts) [inline sepolicy/mediaserver.te]: FYI if/when you update this breaks.

Comment (ceiba1985) [inline sepolicy/mediaserver.te]: @williamcroberts I don't understand your comment, did you mean mediaserver shouldn't have access to surfaceflinger_tmpfs? but I find this rule on all the android_n projects of Intel, Broxton, Cherrytrail etc.

File: androidia_64/AndroidBoard.mk [hunk 2]
```diff
@@ -248,15 +248,20 @@ include device/intel/android_ia/common/audio/AndroidBoard.mk
 ##############################################################
 INSTALLED_CONFIGIMAGE_TARGET := $(PRODUCT_OUT)/config.img
 
-$(INSTALLED_CONFIGIMAGE_TARGET) : $(MKEXTUSERIMG) $(MAKE_EXT4FS) $(E2FSCK)
+selinux_fc := $(TARGET_ROOT_OUT)/file_contexts.bin
+
+$(INSTALLED_CONFIGIMAGE_TARGET) : PRIVATE_SELINUX_FC := $(selinux_fc)
```
Comment (ceiba1985) [inline androidia_64/AndroidBoard.mk]: test

File: androidia_64/AndroidBoard.mk [hunk 3]
```diff
@@ -248,15 +248,20 @@ include device/intel/android_ia/common/audio/AndroidBoard.mk
...
+$(INSTALLED_CONFIGIMAGE_TARGET) : $(MKEXTUSERIMG) $(MAKE_EXT4FS) $(E2FSCK) $(selinux_fc)
```
Comment (ceiba1985) [inline androidia_64/AndroidBoard.mk]: Test

**Comment by kalyankondapally** [comment]
Thanks @sgnanase can you test and push

File: sepolicy/mediaserver.te (hunk 1 above)
Comment (williamcroberts) [inline sepolicy/mediaserver.te]: @ceiba1985 1 proceed with the PR as is, don't consider this comment negative review at all. I talked to the graphics folks, and essentially this is for a control page between clients and surfaceflinger that is shared via passed file descriptor. With that said, IIRC, on N it was required for functionality to work, like adb shell screencap. on AOSP master, if you look at system/sepolicy, the policy is starting to be split, and surfaceflinger_tmpfs becomes a "private" type that we cannot reference in our policy. The graphics team said this functionality is safe to drop in releases after N.

**Comment by williamcroberts** [comment]
@tpalli With regards to writing anywhere to disk, you'll need to have adb in root context or fire up a root shell. After that, you could even call setenforce 0.

**Comment by tpalli** [comment]
@williamcroberts yeah, I'm aware of 'setenforce', I just wanted to highlight that 'system boots to UI' as a testcase is not really enough for this kind of a change (or actually for any change as observed by recent regressions)

**Comment by sgnanase** [comment]
Merged...

**Pull request merged**